web/src/components/vis/PlsdaPage/PlsdaPage.vue

```diff
@@ -0,0 +1,227 @@
+<script>
+import VisTileLarge from '@/components/vis/VisTileLarge.vue';
+import LayoutGrid from '@/components/LayoutGrid.vue';
+import ScorePlot from './ScorePlot.vue';
+import LoadingsPlot from './LoadingsPlot.vue';
+import plotData from '../mixins/plotData';
+
+export default {
+  components: {
+    ScorePlot,
+    LoadingsPlot,
+    VisTileLarge,
+    LayoutGrid,
+  },
+
+  mixins: [plotData('plsda')],
+
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      pcXval: '1',
+      pcYval: '2',
+      numComponentsVal: '10',
+      pcX: 1,
+      pcY: 2,
+      numComponents: 10,
+      showEllipses: true,
+      showCrosshairs: true,
+      showCutoffs: true,
+      showScore: true,
+      showLoadings: true,
+    };
+  },
+
+  computed: {
+    ready() {
+      const pcaReady = this.$store.getters.ready(this.id, 'plsda_scores');
+      const loadingsReady = this.$store.getters.ready(this.id, 'plsda_loadings');
+      return pcaReady && loadingsReady;
+    },
+    loadings() {
+      if (this.plot?.data?.loadings) {
+        return this.plot.data.loadings;
+      }
+      return [];
+    },
+    pcCoords() {
+      if (this.plot?.data?.scores?.x) {
+        return this.plot.data.scores.x;
+      }
+      return [];
+    },
+    eigenvalues() {
+      if (this.plot?.data?.scores?.sdev) {
+        return this.plot.data.scores.sdev;
+      }
+      return [];
+    },
+    rowLabels() {
+      if (this.dataset?.row?.data) {
+        return this.dataset.row.data.filter((row) => row.row_type === 'sample').map((row) => row.row_name);
+      }
+      return [];
+    },
+    groupLabels() {
+      if (this.dataset?.validatedGroups?.data) {
```

Comment:
In other plots, this is part of the plot response object.  Why is it different here?

Also, could you document what `validatedGroups` is in `store/index.js` ?  I don't know what it is or what it's supposed to be used for.

Reply:
It was different because the PLS-DA endpoint didn't send back the `rows` and `labels` like the PCA one. I just changed it to do that so it's more consistent. I'll look at the `validatedGroups` and document that